### PR TITLE
[chores] Updated Architecture Image

### PR DESCRIPTION
**Pull request by atb00ker**
changes

### CODE REVIEW

**Review by nemesisdesign — COMMENTED** [review]
we can probably avoid mentioning "flat json widget" since it's most probably not something most OpenWISP users will be interested in.

OpenWISP IPAM will be a dependency of OpenWISP Controller to implement support for automatic subnet provisioning and Wireguard.

I think we should show Controller first, then Monitoring, then RADIUS, then Topology.

I shared with you the old SVG from 2017 BTW, in case you would like to copy the style.

File: README.md [hunk 1]
```diff
@@ -27,10 +27,19 @@ Like OpenWISP? Find out how to help us!
...
+This ansible script can install the following components on your server.
+Database, postfix, redis and third party services / modules are omitted for simplicity:
 
-![openwisp2 modules diagram](https://raw.githubusercontent.com/openwisp/ansible-openwisp2/master/docs/openwisp2-modules-diagram.png)
+![openwisp2 modules diagram](docs/ansible-openwisp2-modules-diagram.jpg)
+
+- openwisp-controller: The controller is the main part of the application that handles registeration and configuration of the devices.
+- openwisp-network-topology: (Optional) Helps in visualization of your network by creating a topology of all the connected devices.
+- openwisp-radius: (Optional) Handles your freeradius requirements by giving you and HTTP API endpoint to interact with freeradius instance.
+- openwisp-ipam**: (Optional) Helps in IP allocation and network IP management.
```
[the remark this answers is not in the document]
Comment (atb00ker) [reply README.md]: Done.

File: README.md [hunk 2]
```diff
@@ -27,10 +27,19 @@ Like OpenWISP? Find out how to help us!
...
+- openwisp-firmware-upgrader**: (Optional) Helps in mass upgradation of your network's firmware.
+- openwisp-monitoring**: (Optional) Helps in collection of monitoring data for your device.
```
[the remark this answers is not in the document]
Comment (atb00ker) [reply README.md:36]: Done.

File: README.md [hunk 3]
```diff
@@ -27,10 +27,19 @@ Like OpenWISP? Find out how to help us!
...
+Database, postfix, redis and third party services / modules are omitted for simplicity:
 
-![openwisp2 modules diagram](https://raw.githubusercontent.com/openwisp/ansible-openwisp2/master/docs/openwisp2-modules-diagram.png)
+![openwisp2 modules diagram](docs/ansible-openwisp2-modules-diagram.jpg)
+
+- openwisp-controller: The controller is the main part of the application that handles registeration and configuration of the devices.
+- openwisp-network-topology: (Optional) Helps in visualization of your network by creating a topology of all the connected devices.
+- openwisp-radius: (Optional) Handles your freeradius requirements by giving you and HTTP API endpoint to interact with freeradius instance.
+- openwisp-ipam**: (Optional) Helps in IP allocation and network IP management.
+- openwisp-firmware-upgrader**: (Optional) Helps in mass upgradation of your network's firmware.
```
[the remark this answers is not in the document]
Comment (atb00ker) [reply README.md]: Done.

File: README.md [hunk 4]
```diff
@@ -27,10 +27,19 @@ Like OpenWISP? Find out how to help us!
...
-to know more about what happens under the hood in **OpenWISP 2**.
+This ansible script can install the following components on your server.
+Database, postfix, redis and third party services / modules are omitted for simplicity:
 
-![openwisp2 modules diagram](https://raw.githubusercontent.com/openwisp/ansible-openwisp2/master/docs/openwisp2-modules-diagram.png)
+![openwisp2 modules diagram](docs/ansible-openwisp2-modules-diagram.jpg)
+
+- openwisp-controller: The controller is the main part of the application that handles registeration and configuration of the devices.
+- openwisp-network-topology: (Optional) Helps in visualization of your network by creating a topology of all the connected devices.
+- openwisp-radius: (Optional) Handles your freeradius requirements by giving you and HTTP API endpoint to interact with freeradius instance.
```
[the remark this answers is not in the document]
Comment (atb00ker) [reply README.md]: Done.

**Review by atb00ker — COMMENTED** [review]
> I shared with you the old SVG from 2017 BTW, in case you would like to copy the style.

Ah, I wasn't aware that the image is an SVG otherwise I would have used it instead!

I have updated as per the review and removed openwisp-ipam description; it can be added inside the module features, like the rest of the features. :smile:

**Comment by atb00ker** [comment]
> @pniaps 
> supervisor has celery, celerybeat, daphne and openwisp2, so .... service supervisor start|stop|restart
> i had to do sudo chmod +x /usr/lib/influxdb/scripts/influxd-systemd-start.sh for influxdb.service to

**Review by pandafy — COMMENTED** [review]
Great work @atb00ker! 👏🏼  
I have two questions, both of them are nitpicking: 
1. Is there some colour code for individual block or is it random? 
2. I think we can remove hyphenation from OpenWISP Monitoring and OpenWISP Firmware Upgrader. `openwisp-notifications`, `django-loci`, `django-x509` and `netdiff` are in lowercase. Is that intentional?

**Comment by atb00ker** [comment]
@pandafy 
1. Cleared up colors.
2. That's rather random, I removed the dashes. Do you prefer a specific case?

**Review by nemesisdesign — COMMENTED** [review]
Hey Ajay, network topology and radius are not mandatory .
IPAM is going to be come a dependency of OpenWISP Controller.

OpenWISP Users and OpenWISP utils are dependencies which are used by most modules.

I'd change the legend of the blue/teal to mean that those are base modules uses as dependencies and would change the color of users and utils to be of the same.

I'd try to link all the modules/libraries mentioned.

**Comment by atb00ker** [comment]
> >  Hey Ajay, network topology and radius are not mandatory .
> IPAM is going to be come a dependency of OpenWISP Controller.
> OpenWISP Users and OpenWISP utils are dependencies which are used by most modules.
> I'd change the legend of the blue/teal to mean that those are base modules uses as dependencies and would change the color of users and utils to be of the same.
> I'd try to link all the modules/libraries mentioned.

My bad I just came up with the color coding and did not check properly. Fixed.

**Comment by nemesisdesign** [comment]
I just updated the README with the new architecture image: 25a2f3cabbdae09bf4d11a8ae5d7e3946ea153d4. Thanks!

**Pull request closed without merging**